### PR TITLE
Vapor bootstrap

### PR DESCRIPTION
**Pull request by RuiAAPeres**
### Why?

Initial bootstrap of vapor, so we can start migrating our internal implementation to the public. There is no change in the template generated code

### How?

- `vapor new`
- `vapor xcode`
- updated `.gitginore`
- added `.circleci`
- added `.swift-version`
- added `LinuxMain.swift`

### Are these changes (unit/automation/snapshot) tested?
N/A

PR checklist:

* [x]   I've assigned this PR to myself
* [x]   I've set all the necessary PR's labels

### CODE REVIEW

**Comment by RuiAAPeres** [comment]
~Still missing setting up CircleCi.~

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+          command: swift build
+      - run: 
+          name: Run unit tests
+          command: swift test
+
+workflows:
+  version: 2
+  tests:
+    jobs:
+      - test
```
Comment (dmcrodrigues) [inline .circleci/config.yml]: New line missing.

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// swift-tools-version:4.0
+import PackageDescription
+
+let package = Package(
+    name: "WallE",
```
Comment (dmcrodrigues) [inline Package.swift:5]: It can't be `Wall-E`?

**Review by dmcrodrigues — APPROVED** [review]

File: Package.swift (hunk 2 above)
Comment (RuiAAPeres) [reply Package.swift:5]: I avoided on purpose the addition of `-`.

Comment (dmcrodrigues) [reply Package.swift:5]: LGTM

**Pull request merged**